src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Data/DataLayer.cs

```diff
@@ -0,0 +1,228 @@
+			}
+
+			if (object.ReferenceEquals(source, _currentChangesBuffer.Collection))
+			{
+				return;
+			}
+
+			var context = _layerStrategy.CreateUpdateContext(VisitorType.UpdateCollection, mode);
+			var tracker = _layerStrategy.GetTracker(this, context);
+			var initializer = PrepareUpdate(source, tracker);
+
+			// Schedule to flush the buffers to apply the update
+			// TODO uno : Here we can send the prevent the Schedule and allow a sync execution at a given time
+			//			 Like when the message reached the view
+
+			Schedule(initializer.Complete);
+		}
+
+		/// <summary>
+		/// Updates a child data layer
+		/// </summary>
+		internal DataLayerUpdate PrepareUpdate(IObservableCollection source, IUpdateContext context)
+		{
+			var tracker = _layerStrategy.GetTracker(this, context); // re-use the context of the parent layer
+			var update = PrepareUpdate(source, tracker);
+
+			return update;
+		}
+
+		private DataLayerUpdate PrepareUpdate(IObservableCollection source, ILayerTracker tracker)
+		{
+			var from = _nextChangesBuffer ?? _currentChangesBuffer ?? throw new InvalidOperationException("Invalid state. You must invoke the init first.");
+			var to = _nextChangesBuffer = from.UpdateTo(source);
+
+			// Detects changes between 'from.Collection' and 'source'
+			var update = to.Initialize(tracker);
+
+			// Adds the callbacks needed to properly apply the update
+			var previous = default(DataLayerChangesBuffer);
+			update.Prepend(WillApplyUpdate);
+			update.Append(DidApplyUpdate);
+
+			return update;
+
+			void WillApplyUpdate()
+			{
+				previous = _currentChangesBuffer;
+				if (previous != from)
+				{
+					throw new InvalidOperationException("The previous update was not applied yet!");
+				}
+
+				CurrentSourceChanging?.Invoke(this, new CurrentSourceUpdateEventArgs(previous.Collection, to.Collection));
+
+				_currentChangesBuffer = to;
+			}
+
+			void DidApplyUpdate()
+			{
+				if (previous != from)
+				{
+					throw new InvalidOperationException("Wooups ... all callbacks was not applied properly, and that is not recoverable!");
+				}
+
+				CurrentSourceChanged?.Invoke(this, new CurrentSourceUpdateEventArgs(previous.Collection, to.Collection));
+			}
+		}
+
+		internal IObservableCollectionSnapshot PrepareRemove()
+		{
+			if (_currentChangesBuffer is null)
+			{
+				throw new InvalidOperationException("Invalid state. You must invoke the init first.");
+			}
+
+			return _currentChangesBuffer.Stop();
+		}
+
+#region IBindableCollectionViewSource
+		public event EventHandler<CurrentSourceUpdateEventArgs>? CurrentSourceChanging;
+		public event EventHandler<CurrentSourceUpdateEventArgs>? CurrentSourceChanged;
+
+		public TFacet GetFacet<TFacet>()
+		{
+			var facet = _facets.OfType<TFacet>().FirstOrDefault();
+			if (facet == null)
+			{
+				throw new InvalidOperationException($"Facet '{typeof(TFacet).Name}' is not available on this holder ({GetType().Name}).");
```

Comment:
Can you provide a more actionable message?

Reply:
I tried to find a more explicit error message, but I was not able :/ If the user gets it, it's due to an internal error for which the user cannot do anything, except if is explicit "request a facet" then the message is explicit enough.